### PR TITLE
CASMMON-138: Updates to resolve issues when operator doesn't restore service/cray-bos-etcd-client

### PR DESCRIPTION
**Pull request by mantoanibroadwing**
Updating https://github.com/Cray-HPE/docs-csm/blob/release/1.0/operations/kubernetes/Restore_an_etcd_Cluster_from_a_Backup.md

### CODE REVIEW

File: operations/kubernetes/Restore_an_etcd_Cluster_from_a_Backup.md [hunk 1]
```diff
@@ -62,3 +62,13 @@ A backup of a healthy etcd cluster has been created.
         ncn-w001# kubectl -n services delete etcdrestore.etcd.database.coreos.com/cray-bos-etcd
         etcdrestore.etcd.database.coreos.com "cray-bos-etcd" deleted
         ```
+        
+4. Verify if service/cray-bos-etcd-client was created.
```
Comment (mharding-hpe) [inline operations/kubernetes/Restore_an_etcd_Cluster_from_a_Backup.md:66]:
```suggestion
4. Verify that the `cray-bos-etcd-client` service was created.
```

File: operations/kubernetes/Restore_an_etcd_Cluster_from_a_Backup.md [hunk 2]
```diff
@@ -62,3 +62,13 @@ A backup of a healthy etcd cluster has been created.
...
+
+       ```bash
+       $ kubectl get service -A | grep cray-bos-etcd-client
```
Comment (mharding-hpe) [inline operations/kubernetes/Restore_an_etcd_Cluster_from_a_Backup.md:69]:
```suggestion
       ncn# kubectl get service -n services cray-bos-etcd-client
```

File: operations/kubernetes/Restore_an_etcd_Cluster_from_a_Backup.md [hunk 3]
```diff
@@ -62,3 +62,13 @@ A backup of a healthy etcd cluster has been created.
...
+       $
```
Comment (mharding-hpe) [inline operations/kubernetes/Restore_an_etcd_Cluster_from_a_Backup.md:70]:
````suggestion
       ```

       Example of output showing that the service was created:    

       ```text
       NAME                   TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
       cray-bos-etcd-client   ClusterIP   10.28.248.232   <none>        2379/TCP   2m
````

File: operations/kubernetes/Restore_an_etcd_Cluster_from_a_Backup.md [hunk 4]
```diff
@@ -62,3 +62,13 @@ A backup of a healthy etcd cluster has been created.
...
+       ```
+   
+      1. If service was not created repeat steps 1 to 3 for a new backup restore.
```
Comment (mharding-hpe) [inline operations/kubernetes/Restore_an_etcd_Cluster_from_a_Backup.md:73]:
```suggestion
      If the service was not created, then repeat the procedure for a new backup restore.
```

File: operations/kubernetes/Restore_an_etcd_Cluster_from_a_Backup.md [hunk 5]
```diff
@@ -28,7 +28,7 @@ A backup of a healthy etcd cluster has been created.
     cray-bos/etcd.backup_v86767_2020-03-19-18:00:05
     ```
 
-2.  Restore the cluster using a backup.
+2. Restore the cluster using a backup.
 
     Replace etcd.backup\_v277935\_2020-03-30-23:52:54 in the command below with the name of the backup being used.
```
Comment (mharding-hpe) [inline operations/kubernetes/Restore_an_etcd_Cluster_from_a_Backup.md:33]:
```suggestion
    Replace `etcd.backup_v277935_2020-03-30-23:52:54` in the command below with the name of the backup being used.
```

**Review by mharding-hpe — CHANGES_REQUESTED** [review]

**Comment by johren-hpe** [comment]
I wonder why only two checks out of the 6 have run here.

**Comment by mharding-hpe** [comment]
> I wonder why only two checks out of the 6 have run here.

Good question. @mtupitsyn any idea?

**Comment by mharding-hpe** [comment]
I rebased this and force pushed up the changes -- I want to see if it runs all of the checks this time.

File: operations/kubernetes/Restore_an_etcd_Cluster_from_a_Backup.md [hunk 6]
```diff
@@ -62,3 +62,19 @@ A backup of a healthy etcd cluster has been created.
...
+       ```
+
+       Example of output showing that the service was created:    
+
+       ```text
+       NAME                   TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
+       cray-bos-etcd-client   ClusterIP   10.28.248.232   <none>        2379/TCP   2m
+       ```
+   
+      If the service was not created, then repeat the procedure for a new backup restore.
```
Comment (kimjensen-hpe) [inline operations/kubernetes/Restore_an_etcd_Cluster_from_a_Backup.md:79]:
```suggestion
      If the `etcd-client` service was not created, then repeat the procedure to restore the cluster again.
```

**Review by kimjensen-hpe — CHANGES_REQUESTED** [review]
A PR for release/1.2 and main is also needed.

**Comment by mharding-hpe** [comment]
Since @mantoanibroadwing hadn't touch this one in a while, I went ahead and fixed the linter complaints.

**Comment by mharding-hpe** [comment]
It will still need the backports that Kim mentioned above

**Review by mharding-hpe — APPROVED** [review]

File: operations/kubernetes/Restore_an_etcd_Cluster_from_a_Backup.md [hunk 7]
```diff
@@ -28,37 +32,64 @@ A backup of a healthy etcd cluster has been created.
...
     etcdrestore.etcd.database.coreos.com/cray-bos-etcd created
     ```
 
-3.  Restart the pods for the etcd cluster.
+1. Watch the pods come back online.
+
+    This may take a couple minutes.
+
+    ```bash
+    ncn-w001# kubectl -n services get pod | grep SERVICE_NAME
```
Comment (mharding-hpe) [inline operations/kubernetes/Restore_an_etcd_Cluster_from_a_Backup.md:56]:
```suggestion
    ncn# kubectl -n services get pod | grep SERVICE_NAME
```

File: operations/kubernetes/Restore_an_etcd_Cluster_from_a_Backup.md [hunk 8]
```diff
@@ -28,37 +32,64 @@ A backup of a healthy etcd cluster has been created.
...
+    ```
+
+1. Delete the `EtcdRestore` custom resource.
+
+    This step makes it possible for future restores to occur. Replace the `etcdrestore.etcd.database.coreos.com/cray-bos-etcd` value with the name returned in
+    the earlier step when creating the backup.
 
-    1.  Watch the pods come back online.
+    ```bash
+    ncn-w001# kubectl -n services delete etcdrestore.etcd.database.coreos.com/cray-bos-etcd
```
Comment (mharding-hpe) [inline operations/kubernetes/Restore_an_etcd_Cluster_from_a_Backup.md:73]:
```suggestion
    ncn# kubectl -n services delete etcdrestore.etcd.database.coreos.com/cray-bos-etcd
```

**Review by mharding-hpe — APPROVED** [review]

**Comment by mharding-hpe** [comment]
Looks like the backports will require manual intervention

**Review by rustydb — APPROVED** [review]

**Comment by mantoanibroadwing** [comment]
> A PR for release/1.2 and main is also needed.

Will create it for sure

**Comment by rustydb** [comment]
@kimjensen-hpe is on vacation, otherwise we just need her stamp/re-review. We also need the backports created @mantoanibroadwing .

**Review by kimjensen-hpe — APPROVED** [review]
This looks fine for csm1.0, though we should also backport this for 1.2 and 1.3 as it's still relevant

**Pull request merged**